src/PitWallAcquisitionPlugin/PitWallAcquisitionPlugin/UI/ViewModels/IsApiAvailableCommand.cs

```diff
@@ -32,11 +35,35 @@ public bool CanExecute(object parameter)
 
         public void Execute(object parameter)
         {
-            var isAvailable = statusRepo.IsAvailable(parameter.ToString());
+            BackgroundWorker worker = new BackgroundWorker();
+            worker.WorkerReportsProgress = true;
 
-            viewModel.IsApiAvailable = isAvailable;
-        }
+            worker.DoWork += (o, e) =>
```

Comment:
issue(non-blocking) : too much mandatory complexity here. Background worker is pulling a lot of technical stuff in a class that is in the UI layer.

Consider to encapsulate the worker.